### PR TITLE
feat: choose locale in settings

### PR DESCRIPTION
**Pull request by shreyas1599**
#143

### CODE REVIEW

File: lib/app.dart [hunk 1]
```diff
@@ -27,6 +27,24 @@ class MyApp extends StatelessWidget {
     return supportedLocales.first;
   }
 
+  // locale format: {languageCode}_{scriptCode}_{countryCode}
+  static Locale parseLocale(String locale) {
```
Comment (pd4d10) [inline lib/app.dart:31]: Is there any built-in alternative?

Perhaps we should use the built-in parse function (if any) for better edge case handling.

File: lib/models/theme.dart [hunk 2]
```diff
@@ -16,6 +16,31 @@ class DialogOption<T> {
...
+    SupportedLocales.pt_BR,
+    SupportedLocales.zh_Hans,
+  ];
+  static const Map<String, String> languageNameExpanded = {
+    'en': 'English',
+    'hi': 'Hindi',
+    'es': 'Spanish',
+    'nb_NO': 'Norwegian Bokmål (Norway)',
+    'pt_BR': 'Portuguese (Brazil)',
+    'zh_Hans': 'Chinese (Simplified Han)'
```
Comment (pd4d10) [inline lib/models/theme.dart:40]: Should the corresponding text for each language be written in that language?

For example, Spanish -> Español

Comment (pd4d10) [reply lib/models/theme.dart:40]: Perhaps Google translate could help

**Comment by shreyas1599** [comment]
Found a parser. But it's present in a different library. The `Locale` class of the default(`sky_engine`) is different from the `Locale` class of the `intl` library I used. Hence I'm not assigning parsedLocale in the file directly to the locale prop.

**Pull request merged**